infra/events/src/main/java/com/simprints/infra/events/session/SessionEventRepositoryImpl.kt

```diff
@@ -15,8 +18,11 @@ internal class SessionEventRepositoryImpl @Inject constructor(
     private val sessionDataCache: SessionDataCache,
 ) : SessionEventRepository {
 
-    override suspend fun createSession(): EventScope {
+    private val eventsLock = Mutex()
+
+    override suspend fun createSession(): EventScope = eventsLock.withLock {
         closeAllSessions(EventScopeEndCause.NEW_SESSION)
+        Simber.w("Creating session from DB")
```

Comment:
I think this will create too much noise

Reply:
Can we instead log only when we reach the call on line 54:
`?: createSession()`